### DISCUSSION TITLE
Access specific version of Mermaid Documentation and Live Editor

### DISCUSSION THREAD

**Discussion by whitenoisedev** [mermaid-js/mermaid#4348] [General]
It would be nice to have access to the Mermaid Documentation and the Live Editor relative to a specific version.

This is particularly important for apps that bundle mermaid, but not necessarily the latest version, given the frequent addition and/or changes in the diagram syntax.

**Comment by rkraneis** [comment]
Just having a concise changelog that just higlights spec changes would already help. I'm running into "Syntax error in text - mermaid version 10.7.0" all the time with GitLab :/

**Reply by Kenneth-Sills** [reply]
The wayback machine has some captures that can help, @rkraneis. [Here is a capture](https://web.archive.org/web/20240129143815/https://mermaid.js.org/syntax/flowchart.html) from January 29th, 2024 - between the release of 10.7 and 10.8.

Unfortunately, the images aren't preserved.

**Comment by prosoitos** [comment]
Yes for documentation of all past versions!!! (as most documentations do).

I use Mermaid through [Quarto](https://quarto.org/) and they are always running behind. It would be nice to know what will work and what won't without having to do endless trial and error or digging through several change logs.

Thanks.

**Comment by CJPoll** [comment]
+1. I have tools that use older versions of mermaid, and it would be helpful to feed an LLM documentation on which version of mermaid to use.

**Comment by Bengt** [comment]
I need to feed my agent a version of the docs that work with my GitLab wiki.

Versioning documentation seems like a solved problem. readthedocs had a version selector on their pages for ages. What is blocking this?

**Reply by Kenneth-Sills** [reply]
Mermaid uses Vitepress to generate their documentation website, which [does not support versioning natively](https://github.com/vuejs/vitepress/issues/109). As a result, they'd need to either:

1. Keep multiple copies of documentation for old versions in-repo.
2. Reconfigure their current workflows to build and host separate copies for different versions.
3. Switch from Vitepress to a different documentation provider that _does_ support versioning natively.

They're more than happy to [welcome contributors](https://github.com/mermaid-js/mermaid/issues/866), the project has a bit of a backlog and they've had documentation overhaul on the mind for years. Some examples:

- https://github.com/mermaid-js/mermaid/issues/4382
- https://github.com/mermaid-js/mermaid/issues/2910
